integration_test.go

```diff
@@ -0,0 +1,207 @@
+	}
+	dataset := client.Dataset(datasetID)
+	table := dataset.Table(tableID)
+	if err := table.Create(ctx, &bigquery.TableMetadata{Schema: tableSchema}); err != nil {
+		t.Fatalf("Failed to create BigQuery table: %v", err)
+	}
+
+	//Create config file with random table name
+	if err := createConfigFile(tableID); err != nil {
+		t.Fatalf("failed to create config file: %v", err)
+	}
+
+	// Create log file
+	file, err := os.Create(logFilePath)
+	if err != nil {
+		t.Fatalf("Failed to create log file: %v", err)
+	}
+	defer file.Close()
+
+	// Start Fluent Bit with the config file
+	FBcmd := exec.Command("fluent-bit", "-c", "fluent-bit.conf")
+	if err := FBcmd.Start(); err != nil {
+		t.Fatalf("Failed to start Fluent Bit: %v", err)
+	}
+
+	//Wait for fluent-bit connection to generate data; add delays before ending fluent-bit process
+	time.Sleep(2 * time.Second)
+	if err := generateData(10); err != nil {
+		t.Fatalf("Failed to generate data: %v", err)
+	}
+	time.Sleep(2 * time.Second)
+
+	// Stop Fluent Bit
+	if err := FBcmd.Process.Kill(); err != nil {
+		t.Fatalf("Failed to stop Fluent Bit: %v", err)
+	}
+
+	// Verify data in BigQuery by querying
+	queryMsg := "SELECT Message FROM `bigquerytestdefault.siddag_summer2024." + tableID + "`"
+	BQquery := client.Query(queryMsg)
+	BQdata, err := BQquery.Read(ctx)
+	if err != nil {
+		t.Fatalf("Failed to query data information BigQuery: %v", err)
+	}
+
+	rowCount := 0
+	for {
+		//Check that the data sent is correct
+		var BQvalues []bigquery.Value
+		err := BQdata.Next(&BQvalues)
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			t.Fatalf("Failed to read query results: %v", err)
+		}
+
+		assert.Equal(t, "hello world", BQvalues[0])
+		rowCount++
+	}
+
+	// Verify the number of rows
+	assert.Equal(t, 10, rowCount)
+
+	// Clean up - delete the BigQuery table
+	if err := table.Delete(ctx); err != nil {
+		t.Fatalf("Failed to delete BigQuery table: %v", err)
+	}
+
+	// Clean up - delete the log file
+	if err := os.Remove(logFilePath); err != nil {
+		t.Fatalf("Failed to delete log file: %v", err)
+	}
+
+	// Clean up - delete the config file
+	if err := os.Remove(configFilePath); err != nil {
+		t.Fatalf("Failed to delete log file: %v", err)
+	}
+}
+
+// generate a random string with length 10 (to act as a hash for the table name)
+func randString() string {
+	source := rand.NewSource(time.Now().UnixNano())
+	rng := rand.New(source)
+	charset := "qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM0123456789"
+	str := make([]byte, 10)
+	for i := range str {
+		str[i] = charset[rng.Intn(len(charset))]
+	}
+	return string(str)
+}
+
+// template for the config file - dynamically update TableId field
+const configTemplate = `
+[SERVICE]
+    Daemon          off
+    Log_Level       error
+    Parsers_File    ./jsonparser.conf
+    plugins_file    ./plugins.conf
+[INPUT]
+    Name                               tail
+    Path                               ./logfile.log
```

Comment:
Can we make this path to the log file be a parameter too?

Reply:
What do you mean by parameter? Like make the createConfigFile and generateData functions take the path as a parameter? I'm confused as to why we would do this since the test actually generates the log file (and then deletes it). So it's not like there is an existing log file that must be within the directory (I just named it the same as the one in the app sub directory).

Reply:
If the same path is used in more than one place it should not be hardcoded.

Reply:
I can make the log file and configuration file constants. Would this work?

Reply:
Done.